semgrep-core/parsing/Parse_kotlin_tree_sitter.ml

```diff
@@ -1369,25 +1371,30 @@ and primary_constructor (env : env) ((v1, v2) : CST.primary_constructor) =
 and primary_expression (env : env) (x : CST.primary_expression) : expr =
   (match x with
   | `Paren_exp x -> parenthesized_expression env x
-  | `Simple_id x -> 
+  | `Simple_id x ->
       let id = simple_identifier env x in
       Id(id, empty_id_info())
   | `Lit_cst x -> L (literal_constant env x)
   | `Str_lit x -> L (String (string_literal env x))
   | `Call_ref (v1, v2, v3) ->
       let v1 =
         (match v1 with
-        | Some x -> simple_identifier env x
-        | None -> todo env ())
+        | Some x ->
+            let id = simple_identifier env x in
+            Id(id, empty_id_info())
+        | None ->
+            let fake_id = ("None", fake "None") in
```

Comment:
better to keep the todo than using fake thing when you're not sure.

Reply:
yep, probably can add Is and Isnot; I have no idea what it is though. Is it a type check?
maybe we could convert it in an InstanceOf which is part of the 'special' type.

Reply:
yep, they're type checks -- if we convert is and isnot both to instanceof, would we need to differentiate between if it's isnot or is when doing matching?

Reply:
In theory you can create an Ast with a regular Not before the InstanceOf. But problably simpler
you add Is and IsNot in AST_generic.operator.

Reply:
got it! I'll add it to the AST_generic then 😄